Some small fixes to `DB.StateMachine` module.

jonathanknowles opened the pull request:
# Issue Number

Issue #154 
PR #259 

# Overview

This PR makes a number of small fixes to the `Cardano.Wallet.DB.StateMachine` module:

* Fixes bug where the incorrect tag was returned from `createThreeWallets`. 
* Uses consistent naming for maps within `Mock`.
* Fixes some small typos in Haddock comments.
* Extracts out some repetitive function applications.
* Inlines arguments to `readAfterDelete` (since it was always called with the same arguments).

Anviking commented:
(I’ll sanity check this tomorrow)

KtorZ commented on src/Cardano/Wallet/Server/Plugins.hs:
```diff
@@ -102,7 +104,7 @@ apiServer
     let diffusion' = Kernel.fromDiffusion (lower env) diffusion
     logInfo "Testing node client connection"
-    eresp <- liftIO . NodeStateAdaptor.retrying . runExceptT $ NodeClient.getNodeSettings nodeClient
+    eresp <- liftIO . retrying . runExceptT $ NodeClient.getNodeSettings nodeClient
```
What's the need for this now :thinking: ?

Anviking replied:
The [`node-settings`](https://github.com/input-output-hk/cardano-wallet/blob/171f15e4f4de0146f6d1acca70e3b5337cb396d0/src/Cardano/Wallet/API/V1/Handlers/Settings.hs#L18-L28) and `node-info` endpoints where we redirect to the node

KtorZ replied:
I thought that was the goal of this PR to remove these calls :thinking: ?

Anviking replied:
Hm… I was more concerned about the internal uses of NodeStateAdaptor fields. Now these use `withNodeState` again.

KtorZ replied:
Okay. Fine by me.

Anviking commented on src/Cardano/Wallet/Kernel/NodeStateAdaptor.hs:
```diff
@@ -259,6 +257,27 @@ data NodeStateAdaptor m = Adaptor {
+      --
+      -- The latter is simply the node local tip, i.e. "how far we went into
+      -- chasing the global blockchain height during syncing".
+    , getNodeSyncProgress :: LockContext -> m (Maybe BlockCount, BlockCount)
```
We could have `getNodeSyncProgress` remain removed, I guess, but ¯\_(ツ)_/¯

jonathanknowles committed:
Use plural name for map of private keys within `Mock` data type.

The `Mock` data type has the convention that each map has a plural name
based on the type of value held within the map. For consistency with
this convention, the map of wallet ids to private keys should also have
a plural name.

Similarly, we can use `pks` to represent a map, but `pk` to represent a
value within a map.

jonathanknowles committed:
Avoid repetition by moving application of `fmap At`.

jonathanknowles committed:
Use standard indentation (4 spaces).

jonathanknowles committed:
Wrap code to standard 80 characters.

jonathanknowles committed:
Fix bug where the incorrect tag was returned from `createThreeWallets`.

It previously returned `CreateWalletTwice`, but it should return
`CreateThreeWallets`.

jonathanknowles committed:
Inline arguments to function `readAfterDelete`.

This function is always called with the same arguments, so we can safely
inline these arguments.

jonathanknowles committed:
Add missing space.

jonathanknowles committed:
Fix typo and add full stops to Haddock comments.

jonathanknowles committed:
Delete obsolete comment.

KtorZ commented on lib/core/test/unit/Cardano/Wallet/DB/StateMachine.hs:
```diff
@@ -453,7 +453,6 @@ lockstep m@(Model _ ws) c (At resp) = Event
   Generator
 -------------------------------------------------------------------------------}
 
--- NOTE 'concat' reads better.
```
Note that this comment wasn't obsolete. It gives a justification for disabling `hlint` below. Hlint suggests to replace use of `concat` with `++`, but `concat` reads better :)

KtorZ replied:
cc @jonathanknowles ^^
